### PR TITLE
Inspector tabs

### LINKED ISSUE #6118: inspector views tabs
**Summary**
Inspector views are getting more complex and there are more of them. We need a better way to organize them.

**Describe the solution you'd like**
TABS!

**Describe alternatives you've considered**
Accordions

**Testing Instructions**
1. put the inspector through its paces
    2. select different objects in the tree and see that the inspector displays context for the selection
    3. select different tabs and see that the inspector view changes appropriately
    4. go into and out of edit mode (both saving and exiting without saving)
    5. tabs should only be visible for intended selections and in correct state (edit mode, stylable, etc.)
1. some more specific tests
    2. plots have different view / edit inspector views
    3. folders and non-persistables should not display a styles tab
1. things to keep an eye out for
    2. wacky or super long tab names
    3. empty inspector views

### PR DESCRIPTION
**Pull request by davetsay**
Closes #6118

### Describe your changes:
* register all hardcoded inspector views with inspectorViewRegistry
* views get their own tabs
* tabs have logic to hide/show based on `selection` and `isEditing` (only for now)

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Review by ozyx — CHANGES_REQUESTED** [review]
Just a couple minor comments. This is looking really awesome.

File: src/plugins/filters/FiltersInspectorViewProvider.js [hunk 1]
```diff
@@ -31,13 +31,9 @@ define([
...
             key: 'filters-inspector',
-            name: 'Filters Inspector View',
+            name: 'Filters',
             canView: function (selection) {
-                if (selection.length === 0 || selection[0].length === 0) {
-                    return false;
-                }
-
-                let object = selection[0][0].context.item;
+                const object = selection?.[0]?.[0]?.context?.item;
```
[the remark this answers is not in the document]
Comment (davetsay) [reply src/plugins/filters/FiltersInspectorViewProvider.js]: i love how you clean up code you didn't write @ozyx .

Done!

File: src/plugins/inspectorViews/elements/ElementsPool.vue [hunk 2]
```diff
@@ -117,8 +119,8 @@ export default {
                 this.compositionUnlistener();
             }
 
-            if (this.parentObject) {
-                this.composition = this.openmct.composition.get(this.parentObject);
+            if (this.domainObject) {
```
[the remark this answers is not in the document]
Comment (davetsay) [reply src/plugins/inspectorViews/elements/ElementsPool.vue:120]: good catch. left over from the refactor. removed.

File: src/plugins/filters/FiltersInspectorViewProvider.js [hunk 3]
```diff
@@ -53,6 +55,12 @@ define([
                             template: '<filters-view></filters-view>'
                         });
                     },
+                    showTab: function (isEditing) {
+                        const hasPersistedFilters = Boolean(domainObject?.configuration?.filters);
+                        const hasGlobalFilters = Boolean(domainObject?.configuration?.globalFilters);
+
+                        return hasPersistedFilters || hasGlobalFilters;
```
Comment (ozyx) [inline src/plugins/filters/FiltersInspectorViewProvider.js:62]: nice!

**Review by ozyx — APPROVED** [review]
LGTM!

**Review by charlesh88 — COMMENTED** [review]
- [ ] @charlesh88 to do some visual sweetening on the tabs.

**Comment by charlesh88** [comment]
HIGH PRIORITY: when editing a layout or something else with style-able sub-objects, do not switch the tab when a user clicks something else - this is a must-have. Secondarily, I think this should be case for anything, like navigating amongst domain objects.

We need to keep the current tab key in memory, and if a newly selected object also supports that key, don’t switch - only switch if the new object doesn’t have that key.

**Comment by davetsay** [comment]
this should be good to go again. I just need to confirm with you that merge conflict resolution didn't bork annotations, @scottbell.

**Comment by davetsay** [comment]
@scottbell , when you get a chance. I'll need some help in knowing if I borked annotations

File: src/plugins/inspectorViews/annotations/AnnotationEditor.vue [hunk 4]
```diff
@@ -0,0 +1,83 @@
+/*****************************************************************************
```
Comment (scottbell) [inline src/plugins/inspectorViews/annotations/AnnotationEditor.vue]: We can remove this file until we get text annotations implemented.

**Review by scottbell — APPROVED** [review]
Tested tags against notebooks, stacked plots, overlay plots, and search - looks good, Dave!

**Pull request merged**